lib/Doctrine/DBAL/Platforms/AbstractPlatform.php

```diff
@@ -325,7 +327,17 @@ public function getBinaryTypeDeclarationSQL(array $field)
 
         $fixed = $field['fixed'] ?? false;
 
-        if ($field['length'] > $this->getBinaryMaxLength()) {
+        $maxLength = $this->getBinaryMaxLength();
+
+        if ($field['length'] > $maxLength) {
+            if ($maxLength > 0) {
+                @trigger_error(sprintf(
+                    'Binary field length %d is greater than supported by the platform (%d)',
```

Comment:
Can we also add a hint here? _"use getBlobTypeDeclarationSQL() instead"_

Reply:
It may be confusing. Besides using a blob, a user may want to reduce the length of the field. Also, most likely, a user is not calling this method directly, it's called based on their entity definitions.

What about: reduce the field size or use a BLOB field instead?

Reply:
Sounds good. 👍

Reply:
Hah @Ocramius was too fast. 😂